File: go-sdk/pkg/models/operations/deleteroomsroomidstorage.go
package operations

import (
	"openapi/pkg/models/shared"
)

type DeleteRoomsRoomIDStoragePathParams struct {
	RoomID string `pathParam:"style=simple,explode=false,name=roomId"`
}

type DeleteRoomsRoomIDStorageRequest struct {
	PathParams DeleteRoomsRoomIDStoragePathParams
}

type DeleteRoomsRoomIDStorageResponse struct {
	ContentType string
	Error       *shared.Error
	StatusCode  int64
}

File: typescript-sdk/src/sdk/models/operations/deleteroomsroomidstorage.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class DeleteRoomsRoomIdStoragePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=roomId" })
  roomId: string;
}


export class DeleteRoomsRoomIdStorageRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: DeleteRoomsRoomIdStoragePathParams;
}


export class DeleteRoomsRoomIdStorageResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  error?: shared.ErrorT;

  @SpeakeasyMetadata()
  statusCode: number;
}
